kratos/sources/accessor.cpp

```diff
@@ -0,0 +1,166 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Alejandro Cornejo
+//                   Riccardo Rossi
+//                   Carlos Roig
+//
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/accessor.h"
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Classes
+///@{
+
+    /**
+     * @brief This method implements the way to retrieve a double type variable
+     */
+    double const& Accessor::GetProperty(
+        const Variable<double> &rVariable,
+        const Properties &rProperties,
+        const GeometryType &rGeometry,
+        const Vector &rShapeFunctionVector,
+        const ProcessInfo &rProcessInfo
+        ) const
+    {
+        KRATOS_ERROR << "You are calling the GetProperty of the virtual Accessor class..." << std::endl;
+        double aux;
+        return aux;
```

Comment:
1) `aux` is constructed
2) a reference to the locally constructed `aux` is returned
3) => dangling reference

this is true for every default implementation in this class. I'd just recommend turning them into pure virtual functions instead.